src/averageAges.js

```diff
@@ -52,7 +77,23 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const mom = people.filter(person => person.sex === 'f'
+    && people.some(x => x.mother === person.name));
+  let child;
```

Comment:
just assign filtered array here and use different condition for filter method depend on onlyWithSon parameter